### PR TITLE
FEC-11257 - add NPE protection in case of unexpected null session or null url

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java [hunk 1]
```diff
@@ -157,6 +158,10 @@ public void apply(PKMediaEntry mediaEntry, PKMediaEntryInterceptor.Listener list
             // Start the session and get the final stream URL
             session = SmartLib.getInstance().createStreamingSession();
             session.attachPlayer(player, messageBus);
+            if (session == null) {
```
Comment (OrenMe) [inline broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java:167]: If session is null won't it also throw for calling attachPlayer?

File: broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java [hunk 2]
```diff
@@ -146,7 +146,8 @@ public void apply(PKMediaEntry mediaEntry, PKMediaEntryInterceptor.Listener list
         String errorMessage = BroadpeakError.Unknown.errorMessage;
 
         if (mediaEntry != null && mediaEntry.getSources() != null &&
-                !mediaEntry.getSources().isEmpty() && mediaEntry.getSources().get(0) != null) {
+                !mediaEntry.getSources().isEmpty() && mediaEntry.getSources().get(0) != null &&
```
Comment (OrenMe) [inline broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java:156]: Isn't the real issue the fact we have a source without a url? Maybe we need to solve this in the providers where we can validate for sources without url.

Comment (giladna) [reply broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java:156]: @OrenMe  after looking again at the exception it is related to session becoming null not the url

File: broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java (hunk 1 above)
Comment (giladna) [reply broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java:167]: indeed
that i also not sure how exception happened

**Review by OrenMe — APPROVED** [review]

**Pull request merged**